symfit/core/minimizers.py

```diff
@@ -494,14 +531,103 @@ class NelderMead(ScipyMinimize, BaseMinimizer):
     def method_name(cls):
         return 'Nelder-Mead'
 
-
 class Powell(ScipyMinimize, BaseMinimizer):
     """
     Wrapper around :func:`scipy.optimize.minimize`'s Powell algorithm.
     """
 
+class TrustConstr(ScipyHessianMinimize, ScipyConstrainedMinimize, BoundedMinimizer):
+    """
+    Wrapper around :func:`scipy.optimize.minimize`'s Trust-Constr algorithm.
+    """
+    @classmethod
+    def method_name(cls):
+        return 'trust-constr'
+
+    def _get_jacobian_hessian_strategy(self):
+        """
+        Figure out how to calculate the jacobian and hessian. Will return a
+        tuple describing how best to calculate the jacobian and hessian,
+        repectively. If None, it should be calculated using the available
+        analytical method.
+
+        :return: tuple of jacobian_method, hessian_method
+        """
+        if self.jacobian is not None and self.hessian is None:
+            hessian = 'cs'
+        elif self.jacobian is None and self.hessian is None:
+            jacobian = 'cs'
+            hessian = soBFGS(exception_strategy='damp_update')
+        else:
+            jacobian = None
+            hessian = None
+        return jacobian, hessian
+
+    def scipy_constraints(self, constraints):
+        cons = super(TrustConstr, self).scipy_constraints(constraints)
+        # TODO: If no hessian/jac for this model, assume no hessian/jac for the
+        # constraints, and do things with self._get_jacobian_hessian_strategy
```

Comment:
I'm not sure that is necessary actually, at least not for now. Constraints are `Model` subclasses so they have a jacobian and hessian.

However, if in the future any model could be given as a constraint, including numerical ones, then this does become an issue

Reply:
I agree with that analysis, which is why I put a TODO instead of trying to figure out a solution.